clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1480,10 +1618,46 @@ public void commitAsync(OffsetCommitCallback callback) {
      */
     @Override
     public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+        commitAsync(offsets, callback, false);
+    }
+
+    //This is intended for use by both the original KafkaConsumer and RebalanceKafkaConsumer
+    protected void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback, final boolean isChildConsumer) {
         acquireAndEnsureOpen();
         try {
             log.debug("Committing offsets: {}", offsets);
-            coordinator.commitOffsetsAsync(new HashMap<>(offsets), callback);
+            if (!isChildConsumer && useParallelRebalance) {
+                checkRebalance();
+                final RebalanceKafkaConsumer.OffsetInclusion offsetInclusion = RebalanceKafkaConsumer.getRanges(rebalanceConsumer, offsets);
+                final HashMap<TopicPartition, OffsetAndMetadata> parentConsumerMetadata = offsetInclusion.getParentConsumerMetadata();
+                final HashMap<TopicPartition, OffsetAndMetadata> childConsumerMetadata = offsetInclusion.getChildConsumerMetadata();
+                final long hashCode1 = parentConsumerMetadata.hashCode();
+                final long hashCode2 = childConsumerMetadata.hashCode();
+
+                rebalanceConsumer.setOptionalInputArgument(childConsumerMetadata, hashCode1, hashCode2);
+                rebalanceConsumer.sendRequest(null,
```

Comment:
What if only one of this call and the commitOffsetsAsync call on line 1642 succeeds ?